src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCopyFileFromContainer.groovy

```diff
@@ -57,15 +57,21 @@ class DockerCopyFileFromContainer extends DockerExistingContainer {
 
         def tarStream
         try {
-
             tarStream = containerCommand.exec()
-            def hostDestination = new File(hostPath)
 
-            // if compressed leave file as is otherwise untar
-            if (compressed) {
-                copyFileCompressed(tarStream, hostDestination)
-            } else {
-                copyFile(tarStream, hostDestination)
+            if(onNext) {
+                onNext(tarStream)
+            }
+
+            if(!tarStream.isClosed()) {
```

Comment:
Should we do one or the other here? Meaning if the user has defined an `onNext` closure then we don't attempt to handle anything. Otherwise the user could leave things in an undefined state.